cpp/ql/src/Security/CWE/CWE-732/FilePermissions.qll

```diff
@@ -1,6 +1,29 @@
 import cpp
 import semmle.code.cpp.commons.unix.Constants
 
+bindingset[input]
+int parseHex(string input) {
+  input.prefix(2) = "0x" and
```

Comment:
CodeQL is correct here 😂. `input.prefix(2)` creates a new string at runtime whereas `input.matches("0x%")` won't.

Reply:
Ah, you fixed it already. Excellent!